src/window.ts

```diff
@@ -64,13 +85,32 @@ export function getCachedReferences(window: Window & typeof globalThis): CachedR
     return record;
 }
 
+export function getCachedRedReferences(window: Window & typeof globalThis): CachedRedReferencesRecord {
+    let record: CachedRedReferencesRecord | undefined = WeakMapGet(cachedRedGlobalMap, window);
```

Comment:
I don't think we need to cache the red references, because we use them once during the initialization of the sandbox, and after that, we never ever read the state of the same iframe. Maybe we can just remove this weakmap, and rename the method.